ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcInputFormat.java

```diff
@@ -1727,14 +1727,15 @@ private void populateAndCacheStripeDetails() throws IOException {
     private long computeProjectionSize(List<OrcProto.Type> fileTypes,
         List<OrcProto.ColumnStatistics> stats, boolean[] fileIncluded) throws FileFormatException {
       List<Integer> internalColIds = Lists.newArrayList();
+      int rootColumn = 0;
```

Comment:
rootColumn doesn't seem to change, which is strange, I guess this is more like a constant instead

Reply:
it's not supposed to, it's an index of the root struct element that should be skipped